### ISSUE TITLE
automate collecting changes from gpac -genmd

### ISSUE THREAD

**Issue by nlsdvl** [gpac/wiki#7]
Part of the documentation comes from running the `gpac -genmd` documentation. 
Currently this needs to be updated manually.

**Comment by jeanlf** [comment]
Also, adding new filters will require adding the .md to the git repo and patching the mkdocs tree, it would be nice if we could automate this part as well...

**Comment by nlsdvl** [comment]
Updating the navigation tree, do you mean to update the list of filters only ? How was that implemented on the github wiki ?

the following needs to be taking into account:
- the tree is configured manually in mkdocs.yml, only a portion of the tree is referencing auto generated content
-  the navigation tree takes the file path, but also needs a display/name for the menu. For filters, the filter name is enough. The existing navigation tree items should preserve their current display/name.

Generating the mardown files by running genmd could be done with a github action on the gpac repo. Alternatively cron job could handle it on a daily basis.

**Comment by jeanlf** [comment]
> Updating the navigation tree, do you mean to update the list of filters only ? How was that implemented on the github wiki ?

Yes only for the list of filters. The github wiki sidebar was generated along with the filter md files. This way we were sure to never miss a filter in the navigation bar.

**Comment by jeanlf** [comment]
current status:
1. `gpac -genmd` will now patch mkdocs.yml to inject all filters
2. ./scripts/genmd.sh will add all md and mkdocs.yml to git - this obviously needs some rework once integrated in an automated step

**Comment by jeanlf** [comment]
What's the status here ? Is there anything missing in `gpac` or can we now fully automate the wiki generation ?

**Comment by nlsdvl** [comment]
It's mostly a matter of deciding how/where the automation should be implemented. It could be in github workflow - eg. running in docker - or on the buildbot.

Basically, it is a matter of automating:
- using the fresh build of each commit on gpac's master branch
- checkout this repo and run genmd
- if there is a diff, commit and push it here

**Comment by rbouqueau** [comment]
@jeanlf @aureliendavid Buildbot or github Actions ?

**Comment by aureliendavid** [comment]
this seems pretty github actionable to me

we could do it buildbot-side if they are issues doing cross repo stuff in GA but I think it's possible in a github workflow and a good place for it

**Comment by aureliendavid** [comment]
see https://github.com/gpac/gpac/pull/3223 for a possible implementation of this

feel free to comment if no suitable

**Comment by aureliendavid** [comment]
I've merged the gpac PR with the github action to auto push genmd changes to the wiki

I'll try to be vigilant for the next few commits to see if it works properly

I'll close this issue for now but if the github action does some undesirable things don't hesitate to reopen or to open an issue on the gpac side

thanks
